Consolidate account scopes for `LOWER` (index using) username/domain queries

mjankowski opened the pull request:
I was intending to pull out the index-using improvement from https://github.com/mastodon/mastodon/pull/18468#discussion_r880610618 - only to realize that the single-vs-array dynamic is different in these scenarios.

Putting this in as a first pass which consolidates what I think are all the arel-querying  index-using accounts.username and account.domain stuff down into scopes. May try to pull in that linked PR code area in followup.

mjankowski committed:
Consolidate account scopes for lower (index using) username/domain queries

mjankowski committed:
Revert settings source portion

ClearlyClaire reviewed (dismissed):
This looks good, although looking at this part of the code:
- `with_usernames` / `without_blank_username` does not seem useful at all, as the database will never include accounts with blank usernames
- the `AccountFinder` class as a whole does not seem useful either, everything could be done in `find_remote`

mjankowski committed:
Further simplify?!

mjankowski committed:
Remove unused method

mjankowski committed:
Update setting source

mjankowski commented:
> * `with_usernames` / `without_blank_username` does not seem useful at all, as the database will never include accounts with blank usernames

Removed this.

> * the `AccountFinder` class as a whole does not seem useful either, everything could be done in `find_remote`

Removed the class in favor of keeping logic right in that method and just call the scopes. Also restored some usage in SettingSource which I had dropped in a rebase jam-up earlier.

ClearlyClaire reviewed (approved):
Thanks! This is a nice simplification of this code!
